File: lana/src/main/scala/com/financialforce/lana/Display.scala
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */
package com.financialforce.lana

import com.financialforce.lana.display.QuickPickFileOpen
import com.financialforce.lana.runtime.vscode
import com.financialforce.lana.runtime.vscode.{ExtensionContext, TextEditor, Uri, window}

import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue

class Display(extensionContext: ExtensionContext) {
  val pluginName = "Lana"
  private val outputChannel = window.createOutputChannel(pluginName)

  def output(message: String, showChannel: Boolean = false): Unit = {
    if (showChannel)
      outputChannel.show(preserveFocus = true)
    outputChannel.appendLine(message)
  }

  def showInformationMessage(s: String): Unit = vscode.window.showInformationMessage(s)

  def showErrorMessage(s: String): Unit = vscode.window.showErrorMessage(s)

  def showFile(path: String): Unit = {
    vscode.workspace
      .openTextDocument(Uri.file(path))
      .toFuture
      .map(td => vscode.window.showTextDocument(td))
  }

  def getActiveTextEditor: Option[TextEditor] = Option(vscode.window.activeTextEditor) match {
    case Some(undef) => undef.toOption
    case None        => None
  }

  object QuickPickFileOpen {

    def apply(message: String, multiSelect: Boolean): QuickPickFileOpen =
      new QuickPickFileOpen(message, multiSelect)

  }

}

object Display {
  def apply(extensionContext: ExtensionContext): Display = new Display(extensionContext)
}

File: lana/src/Display.ts
/*
 * Copyright (c) 2020 Certinia Inc. All rights reserved.
 */
import { Uri, commands, window } from 'vscode';

import { appName } from './AppSettings.js';

export class Display {
  private outputChannel = window.createOutputChannel(appName);

  output(message: string, showChannel = false) {
    if (showChannel) {
      this.outputChannel.show(true);
    }
    this.outputChannel.appendLine(message);
  }

  showInformationMessage(s: string): void {
    window.showInformationMessage(s);
  }

  showErrorMessage(s: string): void {
    window.showErrorMessage(s);
  }

  showFile(path: string): void {
    commands.executeCommand('vscode.open', Uri.file(path.trim()));
  }
}
